### PR TITLE
[scala-akka-client] Scala Akka client does not support arbitrary query string parameters

### PR DESCRIPTION
**Pull request by ameenhaq**
Scala Akka client `ApiRequest` (mustache file) includes functionality to take Map type query string parameters. The `paramCreation`, has been modified to check the input and serialise the input as Map (when condition is satisfied) with the introduction of a simple conditional statement to check if it `isMap` type.

This PR is a solution for the issue #8606. 

### PR checklist
 
- [✅] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [✅] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [✅] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [✅] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [✅] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

**Comment by wing328** [comment]
Please run the following to update the samples:
```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```

**Comment by chameleon82** [comment]
As there is no generated example could you please create unit test?

File: samples/client/petstore/scala-akka/src/main/scala/org/openapitools/client/api/PetApi.scala [hunk 1]
```diff
@@ -83,7 +83,23 @@ class PetApi(baseUrl: String) {
...
+   *   code 400 :  (Invalid tag value)
+   *
+   * @param filters Pet properties to filter by
+   */
+  def findPetsByArbitraryFilters(filters: Map[String, Any]): ApiRequest[Seq[Pet]] =
+    ApiRequest[Seq[Pet]](ApiMethods.GET, baseUrl, "/pet/findByStatus", "application/json")
+      .withQueryParam(filters)
+      .withSuccessResponse[Seq[Pet]](200)
+      .withErrorResponse[Unit](400)
+
```
Comment (wing328) [inline samples/client/petstore/scala-akka/src/main/scala/org/openapitools/client/api/PetApi.scala:102]: please move this method to `samples/client/petstore/scala-akka/src/test/scala/PetApiTest.scala` instead.

File: samples/client/petstore/scala-akka/src/main/scala/org/openapitools/client/api/PetApi.scala [hunk 2]
```diff
@@ -20,7 +20,7 @@ import org.openapitools.client.core.ApiKeyLocations._
 
 object PetApi {
 
-  def apply(baseUrl: String = "http://petstore.swagger.io/v2") = new PetApi(baseUrl)
+  def apply(baseUrl: String = "https://petstore.swagger.io/v2") = new PetApi(baseUrl)
```
Comment (wing328) [inline samples/client/petstore/scala-akka/src/main/scala/org/openapitools/client/api/PetApi.scala:23]: please revert this change as the local instance of the petstore server doesn't support https

**Comment by wing328** [comment]
merged via https://github.com/OpenAPITools/openapi-generator/pull/17386
